File: Data Structures & Algorithms/maximum-product-subarray/submission-2.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int curr_max=1, curr_min = 1;
        int ans=INT_MIN;
        for(int& x:nums){
            int temp = curr_max;
            curr_max = max({curr_max*x, curr_min*x, x});
            curr_min = min({temp*x, curr_min*x,x});
            ans = max(ans,curr_max);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        s = ""
        n = min(len(_) for _ in strs)
        for i in range(0,n):
            for j in range(1,len(strs)):
                if strs[0][i]!=strs[j][i]:
                    return s
            s+=strs[0][i]
        return s